apps/examples/testcase/le_tc/compression/tc_lzma.c

```diff
@@ -0,0 +1,138 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <crc32.h>
+#include <tinyara/lzma/LzmaLib.h>
+
+unsigned int propsSize = LZMA_PROPS_SIZE;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: tc_2Kb
+ *
+ * Description:
+ *   Check and verify the checksum of a file of size 2KB after compression and decompression
+ *
+ * Returned Value:
+ *   None
+ ****************************************************************************/
+void tc_2Kb(void)
+{
+
+	int i, ret;
+	unsigned int ini_t, after_t;
+	unsigned int writesize = 2048;
```

Comment:
How about using define like `#define BUFFER_SIZE 2048`?

Reply:
We have two cases in the code wherein buffer_size is 2048 in one and 4096 in the other case. Hence declaration is done separately in each function.

Reply:
Made the changes